File: ai/src/common/Defs.py
from typing import TypeVar, Union, Tuple

X = TypeVar('X')
O = TypeVar('O')
Draw = TypeVar('Draw')
E = TypeVar('E')

Player = Union[X, O]
Cell = Union[Player, E]
Winner = Union[Player, Draw]

Move2D = Tuple[Player, int, int]
Move3D = Tuple[Player, int, int, int]


def getOtherPlayer(player: Player) -> Player:
    return 'O' if player == 'X' else 'X'

File: ai/misc/practice/ts/Defs.ts
export type Player = 'X' | 'O';
export type Winner = Player | '_';
export type Cell = Player | ' ';

export type Move2D = [Player, number, number];
export type Move3D = [Player, number, number, number];

export const getOtherPlayer = (player: Player): Player => player == 'X' ? 'O' : 'X';
